experimental/dds/tree2/src/core/repair/repairDataBuilder.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fail } from "../../util";
+import { ChangeAtomId } from "../rebase";
+import { FieldKey } from "../schema-stored";
+import { Delta } from "../tree";
+
+/**
+ * A function that associates a change that produces repair data with the detached
+ * field where the repair data should be stored.
+ * @alpha
+ */
+export type RepairDataHandler = (changeId: ChangeAtomId) => FieldKey;
```

Comment:
How about renaming this to `RepairDataLocator`? "Locator" is not perfect either since it seems to imply that the repair data is already there, but it seems like a more specific name for the role it plays. I'm interested in other ideas you might have.

Reply:
How about `RepairDataIndexer`?